js/api/Surveys.js

```diff
@@ -148,6 +169,21 @@ function clearSurveyCache(exclusions) {
     }
 
     realm.write(() => {
+      const expiredSurveysLength = expiredSurveys.length;
+      for (var i = 0; i < expiredSurveysLength; i++) {
+        let forms = realm.objects('Form').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+        let timeTriggers = realm.objects('TimeTrigger').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+        let geofenceTriggers = realm.objects('GeofenceTrigger').filtered(`surveyId= "${expiredSurveys[i].surveyId}"`)
+
+        let formsLength = forms.length;
+        for (var j = 0; j < formsLength; j++) {
+          let questions = realm.objects('Question').filtered(`formId= "${expiredSurveys[i].surveyId}"`)
```

Comment:
I'd be curious to know the result of that benchmark.  Things change quickly in JS land.  I just read that [.map()](https://ryanpcmcquen.org/javascript/2015/10/25/map-vs-foreach-vs-for.html) is the fastest way to iterate vs. forEach and for...  I suppose it was the unused var `j` that I thought was more difficult to read/maintain than performance.